### PR TITLE
Observable drag-n-drop for cards

### PR DESCRIPTION
**Pull request by dmitryRomankov**

### CODE REVIEW

**Comment by dmitryRomankov** [comment]
[Transform arrays with data for drag-n-drop into observables](https://trello.com/c/svt2d7Zj/43-transform-arrays-with-data-for-drag-n-drop-into-observables)

File: client/src/store/cards/cards.reducer.ts [hunk 1]
```diff
@@ -18,7 +18,8 @@ export const cardsReducer = (
       return {
         ...state,
         status: Status.Success,
-        cards: [...state.cards, ...action.payload]
+        myCards: [...state.cards, ...action.payload],
```
Comment (vharadkou) [inline client/src/store/cards/cards.reducer.ts]: myCards: action.payload

File: client/src/store/cards/cards.reducer.ts [hunk 2]
```diff
@@ -35,7 +36,7 @@ export const cardsReducer = (
       return {
         ...state,
         deck: state.deck,
-        myCards: [...state.myCards, ...cardsArrToMe]
+        myCards: [...state.myCards]
```
Comment (vharadkou) [inline client/src/store/cards/cards.reducer.ts]: unnecessary, already have ...state

File: client/src/store/cards/cards.action.ts [hunk 3]
```diff
@@ -1,12 +1,15 @@
 import { Action } from '@ngrx/store';
 import { Card } from 'models';
+import { CdkDragDrop } from '@angular/cdk/drag-drop';
```
Comment (vharadkou) [inline client/src/store/cards/cards.action.ts]: please specify ordered-imports in tslint 
`        "ordered-imports": [
            true,
            {
                "grouped-imports": true,
                "import-sources-order": "lowercase-last",
                "named-imports-order": "lowercase-last",
                "module-source-path": "full"
            }
        ]`

File: client/src/store/cards/cards.reducer.ts [hunk 4]
```diff
@@ -45,15 +46,40 @@ export const cardsReducer = (
         enemyCardCount: state.enemyCardCount + action.payload.amount
       };
 
+    case CardsActionTypes.MoveMyCardsWithinArray:
+      state.myCards.splice(action.payload.currentIndex, 0, state.myCards.splice(action.payload.previousIndex, 1)[0]);
```
Comment (vharadkou) [inline client/src/store/cards/cards.reducer.ts]: prefer to generate new array (may be with spread)

Comment (dmitryRomankov) [reply client/src/store/cards/cards.reducer.ts]: there is no need to return a new array

**Pull request merged**